index.js

```diff
@@ -1,9 +1,49 @@
-import {NativeModules} from 'react-native';
+import {NativeModules, Platform, PermissionsAndroid} from 'react-native';
+
+const isAndroid = Platform.OS === 'android';
+
+const defaultTitle = "Request Camera Permission";
+const defaultMessage = "Grant access to enable calling your phone";
 
 var RNImmediatePhoneCall = {
-  immediatePhoneCall: function(number) {
-        NativeModules.RNImmediatePhoneCall.immediatePhoneCall(number);
+  immediatePhoneCall: async function(number, title = defaultTitle, message = defaultMessage, buttonNegative = 'Cancel', buttonPositive = 'OK') {
+    if (isAndroid) {
+      try {
+        let permission = await PermissionsAndroid.check(PermissionsAndroid.PERMISSIONS.CALL_PHONE);
+        if(!permission) {
+          let granted = await requestCallPermission(title, message, buttonNegative, buttonPositive);
+          if (granted) {
+            return NativeModules.RNImmediatePhoneCall.immediatePhoneCall("411");
+          }
+          return Promise.reject("Permission not granted");
+        }
+      } catch (e) {
+        return Promise.reject(e);
+      }
+    }
+    return NativeModules.RNImmediatePhoneCall.immediatePhoneCall("411");
   }
 };
 
-export default RNImmediatePhoneCall;
+async function requestCallPermission(title, message, buttonNegative, buttonPositive) {
+  try {
+    const granted = await PermissionsAndroid.request(
+      PermissionsAndroid.PERMISSIONS.CALL_PHONE,
+      {
+        title,
+        message,
+        buttonNegative,
+        buttonPositive
+      },
+    );
+    if (granted === PermissionsAndroid.RESULTS.GRANTED) {
+      return true;
+    } else {
+      return false;
+    }
+  } catch (err) {
+    return false;
```

Comment:
I'm not sure about removing the error... This could hide something important to the user that will use this lib.